fdbclient/StorageServerInterface.h

```diff
@@ -278,11 +283,35 @@ struct GetKeyValuesReply : public LoadBalancedReply {
 	}
 };
 
+// Describe how to hop. Assume the keys are encoded as Tuple, hence there are concept of "elements".
+struct HopInfo {
+	// Do GetKeyValues, extract the last suffixLen elements from each keys.
```

Comment:
I think we could make this a bit more efficient with modest additional cost in request size by adding a number of _bytes_ to skip before starting to parse a tuple. This would also cover cases where, for some reason, the entire key isn't a well-formed tuple, but the same secondary index pattern is followed after some point.